controllers/client/grafana_client.go

```diff
@@ -36,19 +36,33 @@ func getAdminCredentials(ctx context.Context, c client.Client, grafana *v1beta1.
 			return credentials, nil
 		}
 
-		// rely on username and password otherwise
-		username, err := GetValueFromSecretKey(ctx, grafana.Spec.External.AdminUser, c, grafana.Namespace)
-		if err != nil {
-			return nil, err
+		switch {
+		case grafana.Spec.External.AdminUser != nil:
+			username, err := GetValueFromSecretKey(ctx, grafana.Spec.External.AdminUser, c, grafana.Namespace)
+			if err != nil {
+				return nil, err
+			}
+			credentials.username = string(username)
+		case grafana.Spec.Config["security"]["admin_user"] != "":
+			credentials.username = grafana.Spec.Config["security"]["admin_user"]
+		default:
+			return nil, fmt.Errorf("authentication undefined, set apiKey or userName for external instance: %s/%s", grafana.Namespace, grafana.Name)
```

Comment:
We should refer to `adminUser` instead of `userName` according to the current json schema.